benches/benches/bevy_ecs/schedule.rs

```diff
@@ -0,0 +1,62 @@
+use bevy_app::App;
+use bevy_ecs::prelude::*;
+use criterion::{criterion_group, criterion_main, Criterion};
+
+criterion_group!(benches, build_schedule);
+criterion_main!(benches);
+
+fn build_schedule(criterion: &mut Criterion) {
+    // empty system
+    fn sys() {}
+
+    // Use multiple different kinds of label to ensure that dynamic dispatch
+    // doesn't somehow get optimized away.
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, SystemLabel)]
+    struct NumLabel(usize);
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, SystemLabel)]
+    struct DummyLabel;
+
+    let mut group = criterion.benchmark_group("build_schedule");
+    group.warm_up_time(std::time::Duration::from_millis(500));
+    group.measurement_time(std::time::Duration::from_secs(15));
+
+    // Method: generate a set of `graph_size` systems which have a One True Ordering.
+    // Add system to the stage with full constraints. Hopefully this should be maximimally
+    // difficult for bevy to figure out.
+    let labels: Vec<_> = (0..1000).map(NumLabel).collect();
+
+    // Benchmark graphs of different sizes.
+    for graph_size in [100, 500, 1000] {
+        // Basic benchmark without constraints.
+        group.bench_function(format!("{graph_size}_schedule_noconstraints"), |bencher| {
+            bencher.iter(|| {
+                let mut app = App::new();
+                for _ in 0..graph_size {
+                    app.add_system(sys);
+                }
+                app.run();
```

Comment:
```suggestion
                app.update();
```

This should be equivalent with the default runner, but app.update() is much clearer that we're only running the first iteration.